src/run.cpp

```diff
@@ -75,15 +75,16 @@ namespace
 cosim::execution load_system_structure(
     const cosim::filesystem::path& path,
     cosim::model_uri_resolver& uriResolver,
-    cosim::time_point startTime)
+    cosim::time_point startTime,
+    std::optional<unsigned int> workerThreadCount)
 {
     if (path.extension() == ".xml" ||
         (cosim::filesystem::is_directory(path) &&
             cosim::filesystem::exists(path / "OspSystemStructure.xml"))) {
         const auto config = cosim::load_osp_config(path, uriResolver);
         auto execution = cosim::execution(
             startTime,
-            std::make_shared<cosim::fixed_step_algorithm>(config.step_size));
+            std::make_shared<cosim::fixed_step_algorithm>(config.step_size, workerThreadCount));
```

Comment:
When `worker-threads` option is not given (default), will this value be 0 (no threads will be created)? I think `max_threads_` value is only set when this argument not is given.

Reply:
You are right. Should be fixed now.